src/Assembler.hs

```diff
@@ -0,0 +1,43 @@
+module Assember(assemble) where
+
+import Datatypes
+import Opcodes
+
+assemble :: [IR] -> [Int]
+assemble instructions = (map assembleInstruction instructions)
+
+assembleInstruction :: IR -> Int
+
+assembleInstruction LoadIR = (hexOpcode Lw)
+
+assembleInstruction StoreIR = (hexOpcode Sw)
+
+assembleInstruction (TwoIR (R reg) (I immediate)  masked) =
+  (hexMask masked) + (hexOpcode Addi) + (hexRt reg) + immediate
+
+assembleInstruction (TwoIR (R reg) (C address) masked) =
+  (hexMask masked) + (hexOpcode Ldc) + (hexRt reg) + address
+  -- Address needs to be modulo 2**16
+
+assembleInstruction (ThreeIR operator (R rd) (R rs) (R rt) masked) =
+  (hexMask masked) + (hexOpcode opcode) + (hexRd rd) + (hexRs rs) + (hexRt rt) + (hexAlufunction opcode)
+    where opcode = case operator of
+                BitwiseAnd -> And
+                BitwiseOr -> Or
+                BitwiseXor -> Xor
+                Plus -> Add
+                Minus -> Sub
+                Multiply -> Mul
+                LessThan -> Slt
+                EqualTo -> Seq
+
+assembleInstruction (ThreeIR op rd (I rs) (R rt) masked)
+  | op == Plus = assembleInstruction (ThreeIR op rd (R rt) (I rs) masked)
```

Comment:
This is taken from the Serializer file https://github.com/aleksanb/hdc/blob/master/src/Serializer.hs#L31 .